server/errors.ts

```diff
@@ -28,6 +28,14 @@ export function InviteRequiredError(
   });
 }
 
+export function DomainNotAllowedError(
+  message = "The domain is not allowed for this team"
+) {
+  return httpErrors(403, message, {
+    id: "domain_not_allowed",
```

Comment:
We also need to add a quick error message that's displayed to the user in `Login/Notices.tsx` when they hit this case, it'll appear above the login buttons.